### ISSUE TITLE
Investigate ipfs encryption errors

### ISSUE THREAD

**Issue by oed** [3box/3box#1404]
```
2021-05-24T13:43:29.131-04:00	2021-05-24T17:43:29.131Z libp2p:websockets:listener:error inbound connection failed to upgrade Error: Error occurred during XX handshake: Value is null
2021-05-24T13:43:29.131-04:00	at Noise.performXXHandshake (/js-ceramic/node_modules/libp2p-noise/dist/src/noise.js:166:19)
2021-05-24T13:43:29.131-04:00	at async Noise.performHandshake (/js-ceramic/node_modules/libp2p-noise/dist/src/noise.js:135:20)
2021-05-24T13:43:29.131-04:00	at async Noise.secureInbound (/js-ceramic/node_modules/libp2p-noise/dist/src/noise.js:87:27)
2021-05-24T13:43:29.131-04:00	at async Upgrader._encryptInbound (/js-ceramic/node_modules/libp2p/src/upgrader.js:359:12)
2021-05-24T13:43:29.131-04:00	at async Upgrader.upgradeInbound (/js-ceramic/node_modules/libp2p/src/upgrader.js:97:11)
2021-05-24T13:43:29.131-04:00	at async EventEmitter.<anonymous> (/js-ceramic/node_modules/libp2p-websockets/src/listener.js:22:14) {
2021-05-24T13:43:29.131-04:00	code: 'ERR_ENCRYPTION_FAILED'
2021-05-24T13:43:29.131-04:00	
```

See the IPFS logs [here](https://us-east-2.console.aws.amazon.com/cloudwatch/home?region=us-east-2#logsV2:log-groups/log-group/$252Fecs$252Fceramic-elp-1-1) and [here](https://us-east-2.console.aws.amazon.com/cloudwatch/home?region=us-east-2#logsV2:log-groups/log-group/$252Fecs$252Fceramic-elp-1-2)

**Comment by ukstv** [comment]
@zachferland Here.

**Comment by zachferland** [comment]
Did you notice these as well
```
2021-05-26T16:59:02.764Z libp2p:websockets:error connection error: connect ECONNREFUSED 127.0.0.1:4003
```
we are not even using port 4003 (or advertising that port it appears), although it is a default

These more than anything implied something is not right here. But seems like there are additional issues, ie if this enc error is related to this, would expect to fail before that error, and would not expect that "value is null", when that should probably be the peer id in the error.

**Comment by ukstv** [comment]
Did not notice 4003 port, but anyway things smell wrong here wrt establishing connection.

**Comment by ukstv** [comment]
TLDR: In general, seems to be transitory error message, i.e. IPFS uses errors as control flow mechanism. For our non-transitory messages: some node tries to connect to ours, and fails.

As far as I tested it, it does not directly affect ability of IPFS nodes to communicate. Record retrieval and pubsub work all right on two connected nodes, that sometimes emit the encryption error. This seems (=hypothesis) like a transitory message signalling failed DH key exchange aka connection encryption. After the message, connection is established all right. I could not really grok what is happening there though. In the logs we could see this errors happening regularly - every minute or so, that might indicate some node tries to connect, but fails to do so. I.e. instead of re-starting connection encryption, it tries over and over with old key material. One thing about figuring what node tries to connect: adding libp2p:persistent-peer-store and libp2p:peer-store:key-book would help to at least identify peer id of the nodes connecting to ours.

Anyway, pinged IPFS team at https://github.com/ipfs/js-ipfs/issues/3702
